File: homeWorkSeven/python/interface.py

class Interface():
    def __init__(self):
        self.user_input = None

    def main_menu(self):
        self.user_input = None
        while self.user_input not in range(1, 10):
            try:
                self.user_input = int(input("Меню выбора: \n" +
                                       "1 - Добавить контакт\n" +
                                       "2 - Удалить контакт\n" +
                                       "3 - Изменить контакт\n" +
                                       "4 - Поиск контакта\n" +
                                       "5 - Показать все контакты\n" +
                                       "6 - Импортировать телефонный справочник\n" +
                                       "7 - Экспортировать телефонный справочник \n" +
                                       "8 - Отчистить телефонный справочник \n" +
                                       "9 - Выход\n" +
                                       "Выберите пункт: "))
            except:
                print("Ошибка введите число!")
                pass
        return self.user_input

    def choose_remove_contact(self):
        self.user_input = None
        while self.user_input not in range(1, 3):
            try:
                self.user_input = int(input("Удаление контакта: \n" +
                "1 - По параметру (Имя/Фамилия/Телефон)\n" +
                "2 - По контакту из списка\n" +
                "Выберите пункт: "))
            except:
                print("Ошибка введите число!")
                pass
        return self.user_input

    def choose_change_contact(self):
        self.user_input = None
        while self.user_input not in range(1, 4):
            try:
                self.user_input = int(input("Изменение контакта: \n" +
                "1 - Изменить имя\n" +
                "2 - Изменить фамилию\n" +
                "3 - Изменить номер телефона\n" +
                "Выберите пункт: "))
            except:
                print("Ошибка введите число!")
                pass
        return self.user_input

    def choose_format(self):
        self.user_input = None
        while self.user_input not in range(1, 3):
            try:
                self.user_input = int(input("В каком формате: \n" +
                "1 - Горизонатльный\n" +
                "2 - Вертикальный\n" +
                "Выберите пункт: "))
            except:
                print("Ошибка введите число!")
                pass
        return self.user_input

    def file_not_found(self):
        self.user_input = None
        while self.user_input not in range(1, 3):
            try:
                self.user_input = int(input("Ошибка файл не найден! Создать файл?: \n" +
                "1 - Да\n" +
                "2 - Нет\n" +
                "Выберите пункт: "))
            except:
                print("Ошибка введите число!")
                pass
        return self.user_input

    def contact_not_found(self):
        print("Ошибка! Контакт не найден!")

    def print_contact_in_list(self,index: int,contact):
        print("Контакт найден!")
        print(f"{index + 1}. {contact.get_name()} {contact.get_surname()} {contact.get_phone_number()}")

    def print_all_contacts(self,contacts):
        print("Список контактов:")
        if len(contacts) == 0:
            print("Список контактов пуст!")
        else:
            for contact in contacts:
                print(f"{(contacts.index(contact)) + 1}. {contact.get_name()} {contact.get_name()} {contact.get_phone_number()}")

    def input_name(self):
        self.user_input = "1"
        while self.user_input.isdigit():
            try:
                self.user_input = input("Введите Имя: ")
            except:
                print("Ошибка введите текст!")
                pass
        return self.user_input

    def input_surname(self):
        self.user_input = "1"
        while self.user_input.isdigit():
            try:
                self.user_input = input("Введите Фамилию: ")
            except:
                print("Ошибка введите текст!")
                pass
        return self.user_input

    def input_phone_number(self):
        while True:
            try:
                self.user_input = int(input("Введите Номер телефона: "))
                return self.user_input
            except:
                print("Пожалуйста введите номер в числовом формате!")
                continue

    def file_name(self):
        self.user_input = "1"
        while self.user_input.isdigit():
            try:
                self.user_input = input("Введите имя файла `example.txt`: ")
            except:
                print("Ошибка введите текст!")
                pass
        return self.user_input

    def contact_parametr(self):
        self.user_input = "1"
        while self.user_input.isdigit():
            try:
                self.user_input = input("Введите параметр: ")
            except:
                print("Ошибка введите текст!")
                pass
        return self.user_input

    def lots_contacts(self):
        print("Ошибка! Найдено много контактов, пожалуйста добавте еще один параметр!")

    def index_contact_in_list(self,book):
        self.print_all_contacts(book.get_all_contacts())
        while True:
            try:
                self.user_input = int(input("Выберите контакт: "))
                return self.user_input
            except:
                print("Пожалуйста введите число!")
                continue

    def clear_console(self):
        [print("\n") for x in range(500)]

    def press_enter(self):
        input("Нажмити \"ENTER\" для продолжения...")

    def contact_successfully_deleted(self):
        print("Контакт успешно удалён!")

    def contact_successfully_input(self):
        print("Контакты успешно добавлены!")

    def contact_successfully_extract(self):
        print("Контакты успешно экспортированы!")

File: homeWorkSeven/java/Interface.java
package homeWorkOne;
//Пользовательский интерфейс
import java.util.ArrayList;
import java.util.Scanner;

public class Interface {

    public static int mainMenu() {
        Scanner console = new Scanner(System.in);
        System.out.println("Меню выбора: \n" +
                "1 - Добавить контакт\n" +
                "2 - Удалить контакт\n" +
                "3 - Изменить контакт\n" +
                "4 - Поиск контакта\n" +
                "5 - Показать все контакты\n" +
                "6 - Импортировать телефонный справочник\n" +
                "7 - Экспортировать телефонный справочник \n" +
                "8 - Отчистить телефонный справочник \n" +
                "9 - Выход\n" +
                "Выберите пункт: ");
        while (!console.hasNextInt() || !console.hasNextLong()) {
            System.out.println("Пожалуйста введите число!");
            console.next();
        }
        int n = console.nextInt();
        return n;
    }

    public static int chooseRemoveContact() {
        Interface.clearConsole();
        Scanner console = new Scanner(System.in);
        System.out.println("Удаление контакта: \n" +
                "1 - По параметру (Имя/Фамилия/Телефон)\n" +
                "2 - По контакту из списка\n" +
                "Выберите пункт: ");
        while (!console.hasNextInt() || !console.hasNextLong()) {
            System.out.println("Пожалуйста введите число!");
            console.next();
        }
        int n = console.nextInt();
        return n;
    }

    public static int chooseChangeContact() {
        Scanner console = new Scanner(System.in);
        System.out.println("Изменение контакта: \n" +
                "1 - Изменить имя\n" +
                "2 - Изменить фамилию\n" +
                "2 - Изменить номер телефона\n" +
                "Выберите пункт: ");
        while (!console.hasNextInt() || !console.hasNextLong()) {
            System.out.println("Пожалуйста введите число!");
            console.next();
        }
        int n = console.nextInt();
        return n;
    }

    public static int chooseFormat() {
        Scanner console = new Scanner(System.in);
        System.out.println("В каком формате: \n" +
                "1 - Горизонатльный\n" +
                "2 - Вертикальный\n" +
                "Выберите пункт: ");
        while (!console.hasNextInt() || !console.hasNextLong()) {
            System.out.println("Пожалуйста введите число!");
            console.next();
        }
        int n = console.nextInt();
        return n;
    }

    public static boolean fileNotFound() {
        Interface.clearConsole();
        Scanner console = new Scanner(System.in);
        System.out.println("Ошибка файл не найден! Создать файл?: \n" +
                "1 - Да\n" +
                "2 - Нет\n" +
                "Выберите пункт: ");
        while (!console.hasNextInt() || !console.hasNextLong()) {
            System.out.println("Пожалуйста введите число!");
            console.next();
        }
        int n = console.nextInt();
        return n == 1 ? true : false;
    }

    public static void contactNotFound() {
        System.out.println("Ошибка! Контакт не найден!");
    }

    public static void printContactInList(long index, Contact contact) {
        System.out.println("Контакт найден!");
        System.out.println(String.format("%d. %s %s %s", index + 1, contact.getName(), contact.getSurname(), contact.getPhoneNumber()));
    }

    public static void printAllContacts(ArrayList<Contact> contacts) {
        System.out.println("Список контактов:");
        if (contacts.toArray().length == 0) {
            System.out.println("Список контактов пуст");
        } else {
            for (Contact contact : contacts) {
                System.out.println(String.format("%d. %s %s %d", contacts.indexOf(contact) + 1, contact.getName(), contact.getSurname(), contact.getPhoneNumber()));
            }
        }
    }

    public static String inputName() {
        System.out.println("Введите Имя: ");
        Scanner console = new Scanner(System.in);
        while (console.hasNextLong()) {
            System.out.println("Пожалуйста введите текст!");
            console.nextLine();
        }
        return console.nextLine();
    }

    public static String inputSurname() {
        System.out.println("Введите Фамилию: ");
        Scanner console = new Scanner(System.in);
        while (console.hasNextLong()) {
            System.out.println("Пожалуйста введите текст!");
            console.nextLine();
        }
        return console.nextLine();
    }

    public static long inputPhoneNumber() {
        System.out.println("Введите Номер телефона: ");
        Scanner console = new Scanner(System.in);
        while (!console.hasNextLong()) {
            System.out.println("Пожалуйста введите номер в числовом формате!");
            console.next();
        }
        return console.nextLong();
    }

    public static String fileName() {
        System.out.println("Введите имя файла `example.txt`: ");
        Scanner console = new Scanner(System.in);
        while (console.hasNextLong()) {
            System.out.println("Пожалуйста введите текст!");
            console.next();
        }
        return console.nextLine();
    }

    public static String contactParametr() {
        System.out.println("Введите параметр: ");
        Scanner console = new Scanner(System.in);
        return console.nextLine();
    }

    public static void lotsContacts() {
        System.out.println("Ошибка! Найдено много контактов, пожалуйста добавте еще один параметр!");
    }

    public static int indexContactInList(PhoneBook book) {
        printAllContacts(book.getAllContacts());
        System.out.println("Выберите контакт: ");
        Scanner console = new Scanner(System.in);
        while (!console.hasNextInt() || !console.hasNextLong()) {
            System.out.println("Пожалуйста введите число!");
            console.next();
        }
        return console.nextInt() - 1;
    }

    public static void clearConsole() {
        for (int clear = 0; clear < 500; clear++) {
            System.out.println("\b");
        }
    }

    public static void pressEnter() {
        System.out.println();
        System.out.println("Нажмити \"ENTER\" для продолжения...");
        Scanner scanner = new Scanner(System.in);
        scanner.nextLine();
    }

    public static void contactSuccessfullyDeleted() {
        System.out.println("Контакт успешно удалён!");
    }

    public static void contactSuccessfullyInput() {
        System.out.println("Контакты успешно добавлены!");
    }

    public static void contactSuccessfullyExtract() {
        System.out.println("Контакты успешно экспортированы!");
    }
}

